src/baseTrie.ts

```diff
@@ -611,12 +618,23 @@ export class Trie {
     const rlpNode = node.serialize()
 
     if (rlpNode.length >= 32 || topLevel) {
-      const hashRoot = node.hash()
-      opStack.push({
-        type: 'put',
-        key: hashRoot,
-        value: rlpNode,
-      })
+      // Do not use TrieNode.hash() here otherwise serialize()
+      // is applied twice (performance)
+      const hashRoot = keccak(rlpNode)
```

Comment:
I was a bit confused here but this is the "unpacked" `node.hash()` function where we just hash this raw `rlpNode` data.